### PR TITLE
[daemon] Modify prompt text for user input. Contributes to JB#45838

### PR DESCRIPTION
**Pull request by skitternd**
- Prompt text now is more clear and brief
- Prompt text template is changed for platform applications

### CODE REVIEW

**Comment by chriadam** [comment]
LGTM, thanks.  Is this required for 3.1.0 / LTS2?  If so, I will need to ping Jarkko about this one.

**Comment by skitternd** [comment]
@chriadam Yes, I think so.

File: daemon/SecretsImpl/secretsrequestprocessor.cpp [hunk 1]
```diff
@@ -1346,10 +1346,18 @@ Daemon::ApiImpl::RequestProcessor::userInput(
...
 
         InteractionParameters::PromptText promptText = promptParams.promptText();
-        //: Inform the user that the application is requesting data. %1 is the application name, %2 is the prompt text supplied by the application.
-        //% "An application %1 is requesting input which will be returned to the application: %2"
-        promptText.setMessage(qtTrId("sailfish_secrets-user_input-la-data_request")
-                              .arg(callerApplicationId, promptText.message()));
+
+        if (applicationIsPlatformApplication) {
+            //: Inform the user that some system application is requesting data. %1 is the prompt text supplied by the application.
+            //% "System application is requesting input: %1"
```
Comment (chriadam) [inline daemon/SecretsImpl/secretsrequestprocessor.cpp]: Perhaps "A system application is requesting input: %1" instead.

**Comment by chriadam** [comment]
Do we need also separate system-application-specific translation strings for all of the other UI strings which currently include the application name?

e.g. a quick grep found the following, but there may be some I missed:

https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L507
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L727
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L742

https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L1508
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L1564
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L2299
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L2349
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L2589
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L3041
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L3380
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L3879
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4433
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4443
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4510
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4522
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4780
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L4790
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L5100
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L5108
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L5116
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L5124
https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/SecretsImpl/secretsrequestprocessor.cpp#L5559

**Comment by skitternd** [comment]
> https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L507
> https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L727
> https://github.com/sailfishos/sailfish-secrets/blob/master/daemon/CryptoImpl/cryptorequestprocessor.cpp#L742

These are not relevant.

The rest I fixed and then some more. Please, check again.

File: daemon/SecretsImpl/secretsrequestprocessor.cpp (hunk 1 above)
Comment (skitternd) [reply daemon/SecretsImpl/secretsrequestprocessor.cpp]: done

File: daemon/SecretsImpl/secretsrequestprocessor.cpp [hunk 2]
```diff
@@ -35,6 +35,10 @@ using namespace Sailfish::Secrets;
 
 namespace {
 
+    //: A string shown to user instead of the application ID if it is a platform application.
+    //% "A system application"
```
Comment (jpetrell) [inline daemon/SecretsImpl/secretsrequestprocessor.cpp]: On UI we talk about "apps" not "applications".

**Comment by jpetrell** [comment]
"Sailfish OS" in security dialogs is meaningful, we want to avoid ambiguity of talking about just "OS" or "system". Some kind of customisation mechanism is needed.

**Comment by chriadam** [comment]
This change (taking OS name from systemsettings) seems good.  But don't we still need some minimisation of the strings presented in the various UI flows (with special casing for system vs non-system cases), as was originally discussed?

**Comment by skitternd** [comment]
> But don't we still need some minimisation of the strings presented in the various UI flows (with special casing for system vs non-system cases), as was originally discussed?

I think, such change can be discussed in another task/pull request.

**Comment by chriadam** [comment]
Ok.  LGTM.

**Comment by skitternd** [comment]
So, will we merge it?

**Comment by rainemak** [comment]
Will you @chriadam take care of this merge this to master (and 3.1.0 as well).

**Comment by chriadam** [comment]
> Will you @chriadam take care of this merge this to master (and 3.1.0 as well).

Done, tagged as 0.2.13 in master and 0.2.11.1 in upgrade-3.1.0

**Pull request merged**